### PR TITLE
New Salt Stack Cheatsheet

### PR DESCRIPTION
**Pull request by k00laidzone**
Would like to add the salt stack cheat sheets.

---
https://duck.co/ia/view/salt_stack_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@k00laidzone Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub ID:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata post coming soon

**Comment by jophab** [comment]
@k00laidzone you could submit only one file at a time.
so remove the other two files.
You can submit pull requests for those two files seperetly from two new branches.

File: share/goodie/cheat_sheets/json/salt.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
+    "id": "salt_cheat_sheet",
+    "name": "Salt",
+    "description": "Common commands for SaltStack Master/Minion",
+    "metadata": {
+        "sourceName": "SaltStack Website",
+        "sourceUrl": "https://docs.saltstack.com/en/latest/ref/cli/"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/salt.json]: @k00laidzone "SaltStack" is enough as the source name.
So lets remove website

File: share/goodie/cheat_sheets/json/salt.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            "key": "salt -d"
+        }, {
+            "val": "Specify the location of the pidfile. Default: /var/run/salt-minion.pid",
+            "key": "salt --pid-file"
+        }],
+        "Other helpful Commands": [{
+            "val": "After setting up the master it's a good idea to tap the even bus to watch what's going on around the network.",
+            "key": "salt-run state.event pretty=True"
+        }]
+    }
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/salt.json]: @k00laidzone please remove periods from end of val fields.

File: share/goodie/cheat_sheets/json/salt.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        }, {
+            "val": "Reject the specified public key (use --include-all to match accepted keys in addition to pending keys). Globs are supported.",
+            "key": "salt-key -r"
+        }, {
+            "val": "Rejects all pending keys.",
+            "key": "salt-key -A"
+        }, {
+            "val": "salt-key -d  <minion_name>",
+            "key": "Delete the specified key. Globs are supported."
+        }, {
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/salt.json]: @k00laidzone  I think key and val fields are flipped here.

File: share/goodie/cheat_sheets/json/salt.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        "sourceName": "SaltStack Website",
+        "sourceUrl": "https://docs.saltstack.com/en/latest/ref/cli/"
+    },
+    "template_type": "terminal",
+    "section_order": [
+        "Format",
+        "salt-key Commands",
+        "salt-minion/master Commands",
+        "Other helpful Commands"
+    ],
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/salt.json]: @k00laidzone lets capitalise h in helpful also

**Comment by moollaza** [comment]
k00laidzone Thanks for contributing and welcome to DuckDuckHack!

I think we should change the name/id of this cheat sheet to `salt_stack_cheat_sheet` and `salt-stack.json`.

Otherwise this will trigger on "salt cheat sheet" which I believe is a little too broad of a trigger for this IA

**Comment by moollaza** [comment]
I've updated the ID of the IA Page already and the link above. I just need you to change the JSON and then I think this is just about done!

File: share/goodie/cheat_sheets/json/salt.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            "key": "salt -u"
+        }, {
+            "val": "Run salt-minion as a daemon",
+            "key": "salt -d"
+        }, {
+            "val": "Specify the location of the pidfile. Default: /var/run/salt-minion.pid",
+            "key": "salt --pid-file"
+        }],
+        "Other helpful commands": [{
+            "val": "After setting up the master it's a good idea to tap the even bus to watch what's going on around the network",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/salt.json]: `even bus` is that a typo?

File: share/goodie/cheat_sheets/json/salt.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            "val": "Print the version of Salt that is running",
+            "key": "salt --version"
+        }, {
+            "val": "Show program's dependencies and version number, and then exit",
+            "key": "salt --versions-report"
+        }, {
+            "val": "Show the help message and exit",
+            "key": "salt -h"
+        }, {
+            "val": "The location of the Salt configuration directory. This directory contains the configuration files for Salt master and minions. The default location on most systems is /etc/salt",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/salt.json]: Let's simplify this a bit:

`"The location of the Salt configuration directory, which contains the configuration files for Salt master and minions. The default location on most systems is /etc/salt",`

File: share/goodie/cheat_sheets/json/salt.json [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    "id": "salt_cheat_sheet",
+    "name": "Salt",
+    "description": "Common commands for SaltStack Master/Minion",
+    "metadata": {
+        "sourceName": "SaltStack",
+        "sourceUrl": "https://docs.saltstack.com/en/latest/ref/cli/"
+    },
+    "template_type": "terminal",
+    "section_order": [
+        "Salt key commands",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/salt.json]: Can we Title Case These? `Salt Key Commands, Salt Minion/Master Commands, Other Commands`

File: share/goodie/cheat_sheets/json/salt.json [hunk 8]
```diff
@@ -0,0 +1,62 @@
+{
+    "id": "salt_cheat_sheet",
+    "name": "Salt",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/salt.json]: Should be `SaltStack`

**Comment by moollaza** [comment]
@k00laidzone Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

BTW, thanks for doing such a great job so far! :cake: We really appreciate you taking the time to contribute. As a thanks, we'd like to write a short blog post about you and your Instant Answer.

If you're interested please send an email to open@duckduckgo.com, with the subject "DuckDuckHack Dev Appreciation".

Please provide your name and include a link to this pull request as well.

We'll get in touch for more details!

**Comment by mintsoft** [comment]
@k00laidzone I'm closing this for now, feel free to ping us and we'll happily reopen it if you want

**Comment by k00laidzone** [comment]
Thank you sorry for the delays.

**Pull request closed without merging**